product_cost_security/models/product_template.py

```diff
@@ -17,3 +20,11 @@ def _compute_user_can_update_cost(self):
             product.user_can_update_cost = self.env.user.has_group(
                 "product_cost_security.group_product_edit_cost"
             )
+
+    @api.depends_context("uid")
+    def _compute_user_can_see_standard_price_uom(self):
+        """A user can see cost only when it has required permissions."""
+        for product in self:
+            product.user_can_see_standard_price_uom = self.env.user.has_group(
+                "product_cost_security.group_product_cost"
+            ) and (product.product_variant_count <= 1 or product.is_product_variant)
```

Comment:
This method should depend on `"product_variant_count"` and `"is_product_variant"`

Reply:
Yes, I will make that change